tcms/static/js/testrun_actions.js

```diff
@@ -259,91 +269,61 @@ function updateRunStatus(object_pk, value, callback) {
   });
 }
 
-var updateCaseRunStatus = function(e) {
-  e.stopPropagation();
-  e.preventDefault();
-  var container = jQ(this).parents().eq(3);
-  var parent = container.parent();
-  var title = parent.prev();
-  var link = title.find('.expandable')[0];
-  var parameters = Nitrate.Utils.formSerialize(this);
-  var object_pk = parameters['object_pk'];
-  var value = parameters['value'];
-
-  // Callback when
-  var callback = function(t) {
-    // Update the contents
-    const value = parameters['value']
-    if (value != '') {
-      // Update the case run status icon
-      var crs = Nitrate.TestRuns.CaseRunStatus;
-      title.find('.execution_status_icon').each(function(index) {
-        for (i in crs) {
-          jQ(this).removeClass(crs[i].icon);
-        }
-        const execution = Nitrate.TestRuns.CaseRunStatus.find(tcs => tcs.pk === value);
-        jQ(this).addClass(execution.icon);
-        jQ(this).css('color', execution.color)
-      });
+function updateExecutionStatus(executionId, caseId, executionStatusPk) {
 
-      // Update related people
-      var usr = Nitrate.User;
-      title.find('.link_tested_by').each(function(i) {
-        jQ(this).html(usr.username);
-      });
-    }
+  const container = jQ('.update_form').parents().eq(3)
+  const parent = container.parent()
+  const title = parent.prev()
+  const link = title.find('.expandable')[0];
+
+  const comment = jQ('td > textarea[name=comment]').val()
+
+  setContainerToLoading(container, caseId)
+
+  if (comment) {
+    jsonRPC('TestExecution.add_comment', [executionId, comment], () => updateCommentsCount(caseId, true));
+  }
+
+  // Update the object when changing the status
+  jsonRPC('TestExecution.update', [executionId, {'status': executionStatusPk}], () => {
+    // Update the case run status icon
+    // TODO: remove this global variable in favour of API call
+    var executionStatuses = Nitrate.TestRuns.CaseRunStatus;
```

Comment:
this function is the only one using this variable so you may as well add a `const executionStatuses` which gets assigned when the page gets ready/onload and uses the RPC method to read the info.

Reply:
done, but it can't be a `const` since a `const` needs to be initialized right away. 
if we want it to be a global variable it should be `let`. The other option is to pass it as an argument, but I think global var (only used in this file) is better